gradle/integration-test.gradle

```diff
@@ -16,4 +16,51 @@ task integrationTest(type: Test) {
   mustRunAfter test
 }
 
+// small batches of ITs to avoid memory errors on CI. will be run in parallel workflows.
+task it1(type: Test) {
```

Comment:
What do you about this shorter version to avoid duplication of code?

```
task integrationTest(type: Test) {
  description = 'Runs the integration tests.'
  group = 'verification'
}

['**/ScanIT_Gradle_Versions_4_2*.class',
 '**/ScanIT_Gradle_Versions_4_5*.class',
 '**/ScanIT_Gradle_Versions_4_8*.class',
 '**/ScanIT_Gradle_Versions_5_1*.class',
 '**/ScanIT_Gradle_Versions_5_4*.class'].eachWithIndex { className, index -> 
  task "it${index + 1}"(type: Test) {
    includes = [className]
    description = "Runs the integration tests on class $className."
    group = 'verification'
    testClassesDirs = sourceSets.integrationTest.output.classesDirs
    classpath = sourceSets.integrationTest.runtimeClasspath
    maxHeapSize = '1536m'
    mustRunAfter test
  }
  
  integrationTest.dependsOn "it${index + 1}"
}
```

Reply:
Wow. Nifty! I bet I messed up the change, but I'm getting the errors below trying to run:

```
$ gradle integrationTest
...
> Task :it1

org.sonatype.gradle.plugins.scan.ScanIT_Gradle_Versions_4_2_to_4_4 > testScanTask_FailPolicyAction_NexusIQ[Version: 4.2.1] FAILED
    java.lang.IllegalStateException
        Caused by: org.gradle.tooling.GradleConnectionException
            Caused by: java.lang.IllegalArgumentException

org.sonatype.gradle.plugins.scan.ScanIT_Gradle_Versions_4_2_to_4_4 > testScanTask_NoPolicyAction_NexusIQ[Version: 4.2.1] FAILED
    java.lang.IllegalStateException
        Caused by: org.gradle.tooling.GradleConnectionException
            Caused by: java.lang.IllegalArgumentException
...
```

Similar issue if I run `$ gradle it1`.

Reply:
> Wow. Nifty! I bet I messed up the change, but I'm getting the errors below trying to run:
> 
> ```
> $ gradle integrationTest
> ...
> > Task :it1
> 
> org.sonatype.gradle.plugins.scan.ScanIT_Gradle_Versions_4_2_to_4_4 > testScanTask_FailPolicyAction_NexusIQ[Version: 4.2.1] FAILED
>     java.lang.IllegalStateException
>         Caused by: org.gradle.tooling.GradleConnectionException
>             Caused by: java.lang.IllegalArgumentException
> 
> org.sonatype.gradle.plugins.scan.ScanIT_Gradle_Versions_4_2_to_4_4 > testScanTask_NoPolicyAction_NexusIQ[Version: 4.2.1] FAILED
>     java.lang.IllegalStateException
>         Caused by: org.gradle.tooling.GradleConnectionException
>             Caused by: java.lang.IllegalArgumentException
> ...
> ```
> 
> Similar issue if I run `$ gradle it1`.

I tested it locally and worked fine. Please check this commit and let me know how it behaves on your machine: cdd22b373bd9cea16e8e0bfc1d0185a93e1b61a9

Reply:
Also I ran `./gradlew integrationTest --info` to make sure the wrapper was used instead of a local Gradle installation.

Reply:
Boggle. I even copy/pasted from the commit you pasted above. Here's a patch of my local changes. I'm still not seeing the difference.

```
Index: gradle/integration-test.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- gradle/integration-test.gradle	(revision 66ab207ef3a132917bc400ffec067e4931bc54df)
+++ gradle/integration-test.gradle	(date 1581116032797)
@@ -6,61 +6,29 @@
     runtimeClasspath += output + compileClasspath
   }
 }
+// small batches of ITs to avoid memory errors on CI. will be run in parallel workflows.
 
 task integrationTest(type: Test) {
   description = 'Runs the integration tests.'
   group = 'verification'
-  testClassesDirs = sourceSets.integrationTest.output.classesDirs
-  classpath = sourceSets.integrationTest.runtimeClasspath
-  maxHeapSize = '1536m'
-  mustRunAfter test
-}
-
-// small batches of ITs to avoid memory errors on CI. will be run in parallel workflows.
-task it1(type: Test) {
-  description = 'Runs small batch of integration tests.'
-  group = 'verification'
-  testClassesDirs = sourceSets.integrationTest.output.classesDirs
-  classpath = sourceSets.integrationTest.runtimeClasspath
-  maxHeapSize = '1536m'
-  includes = ['**/ScanIT_Gradle_Versions_4_2*.class']
-  mustRunAfter test
-}
-task it2(type: Test) {
-  description = 'Runs small batch of integration tests.'
-  group = 'verification'
-  testClassesDirs = sourceSets.integrationTest.output.classesDirs
-  classpath = sourceSets.integrationTest.runtimeClasspath
-  maxHeapSize = '1536m'
-  includes = ['**/ScanIT_Gradle_Versions_4_5*.class']
-  mustRunAfter test
-}
-task it3(type: Test) {
-  description = 'Runs small batch of integration tests.'
-  group = 'verification'
-  testClassesDirs = sourceSets.integrationTest.output.classesDirs
-  classpath = sourceSets.integrationTest.runtimeClasspath
-  maxHeapSize = '1536m'
-  includes = ['**/ScanIT_Gradle_Versions_4_8*.class']
-  mustRunAfter test
 }
-task it4(type: Test) {
-  description = 'Runs small batch of integration tests.'
-  group = 'verification'
-  testClassesDirs = sourceSets.integrationTest.output.classesDirs
-  classpath = sourceSets.integrationTest.runtimeClasspath
-  maxHeapSize = '1536m'
-  includes = ['**/ScanIT_Gradle_Versions_5_1*.class']
-  mustRunAfter test
-}
-task it5(type: Test) {
-  description = 'Runs small batch of integration tests.'
-  group = 'verification'
-  testClassesDirs = sourceSets.integrationTest.output.classesDirs
-  classpath = sourceSets.integrationTest.runtimeClasspath
-  maxHeapSize = '1536m'
-  includes = ['**/ScanIT_Gradle_Versions_5_4*.class']
-  mustRunAfter test
-}
+
+['**/ScanIT_Gradle_Versions_4_2*.class',
+ '**/ScanIT_Gradle_Versions_4_5*.class',
+ '**/ScanIT_Gradle_Versions_4_8*.class',
+ '**/ScanIT_Gradle_Versions_5_1*.class',
+ '**/ScanIT_Gradle_Versions_5_4*.class'].eachWithIndex { className, index ->
+  task "it${index + 1}"(type: Test) {
+    includes = [className]
+    description = "Runs the integration tests on class $className."
+    group = 'verification'
+    testClassesDirs = sourceSets.integrationTest.output.classesDirs
+    classpath = sourceSets.integrationTest.runtimeClasspath
+    maxHeapSize = '1536m'
+    mustRunAfter test
+  }
 
+  integrationTest.dependsOn "it${index + 1}"
+}
+
 check.dependsOn integrationTest
```

Reply:
I wonder if this could be a 'gradle version' thang? I pulled down you changes, and get the same errors.